applications/StructuralMechanicsApplication/custom_constitutive/linear_elastic_orthotropic_2D_law.cpp

```diff
@@ -0,0 +1,422 @@
+		{
+			Matrix& ConstitutiveMatrix = rValues.GetConstitutiveMatrix();
+			this->CalculateLinearElasticMatrix(ConstitutiveMatrix, MaterialProperties);
+		}
+	}
+
+	
+
+	//***********************************UPDATE*******************************************
+	//************************************************************************************
+	/*
+	// pwdebug
+	void LinearElasticOrthotropic2DLaw::FinalizeMaterialResponsePK2(Parameters& rValues)
+	{
+
+		rValues.Set(ConstitutiveLaw::FINALIZE_MATERIAL_RESPONSE);
+		this->CalculateMaterialResponsePK2(rValues);
+		rValues.Reset(ConstitutiveLaw::FINALIZE_MATERIAL_RESPONSE);
+
+		UpdateInternalVariables(rValues);
+	}
+	*/
+
+	//************************************************************************************
+	//************************************************************************************
+	/*
+	// pwdebug
+	void LinearElasticOrthotropic2DLaw::UpdateInternalVariables(Parameters& rValues)
+	{
+		const Matrix& DeformationGradientF = rValues.GetDeformationGradientF();
+		const double& DeterminantF = rValues.GetDeterminantF();
+
+		Matrix DeformationGradientF0 = DeformationGradientF;
+		DeformationGradientF0 = Transform2DTo3D(DeformationGradientF0);
+		MathUtils<double>::InvertMatrix(DeformationGradientF0, this->mInverseDeformationGradientF0, mDeterminantF0);
+		mDeterminantF0 = DeterminantF; //special treatment of the determinant 
+	}
+	*/
+	
+	//***********************COMPUTE TOTAL STRAIN*****************************************
+	//************************************************************************************
+
+	void LinearElasticOrthotropic2DLaw::CalculateGreenLagrangeStrain(const Matrix & rRightCauchyGreen,
+		Vector& rStrainVector)
+	{
+		// TAKEN FROM LinearElasticPlasticPlaneStrain2DLaw
+		//E= 0.5*(FT*F-1)
+		rStrainVector[0] = 0.5 * (rRightCauchyGreen(0, 0) - 1.00);
+		rStrainVector[1] = 0.5 * (rRightCauchyGreen(1, 1) - 1.00);
+		rStrainVector[2] = rRightCauchyGreen(0, 1);
+	}
+
+	//***********************COMPUTE TOTAL STRESS PK2*************************************
+	//************************************************************************************
+
+	void LinearElasticOrthotropic2DLaw::CalculateStress(const Vector & rStrainVector,
+		const Matrix & rConstitutiveMatrix,
+		Vector& rStressVector)
+	{
+		//1.-2nd Piola Kirchhoff StressVector increment
+		if (rStressVector.size() != rStrainVector.size())
+			rStressVector.resize(rStrainVector.size(), false);
+
+		noalias(rStressVector) = prod(rConstitutiveMatrix, rStrainVector);
+	}
+
+	//***********************COMPUTE ALGORITHMIC CONSTITUTIVE MATRIX**********************
+	//************************************************************************************
+
+	void LinearElasticOrthotropic2DLaw::CalculateLinearElasticMatrix(Matrix& rConstitutiveMatrix,
+		const Properties& rMaterialProperties)
+	{
+		double E1 = rMaterialProperties[YOUNG_MODULUS_X];
```

Comment:
I wasn't entirely sure what you mean by 'at once', but it's been modified such that the values aren't stored in an unneeded intermediate step.